### PR TITLE
Separated Extended and Normal button

### PR DESCRIPTION
**Pull request by TomWayne98**

### CODE REVIEW

File: library/src/main/java/com/github/clans/fab/ExtendedButtonShadowShape.java [hunk 1]
```diff
@@ -28,6 +25,9 @@
 
     @Override
     public void draw(Canvas canvas, Paint paint) {
+        float mExtraLeftShadow = -5f;
```
Comment (davidvavra) [inline library/src/main/java/com/github/clans/fab/ExtendedButtonShadowShape.java]: Don't change this. Is more efficient to define variables in class than in every draw method call.

File: library/src/main/java/com/github/clans/fab/ExtendedFloatingActionButton.java [hunk 2]
```diff
@@ -0,0 +1,1316 @@
...
+/**
+ * Menu button class for extended version of FloatingActionMenu
+ *
+ * @author Josef Hruška (josef@stepuplabs.io)
+ */
+
+public class ExtendedFloatingActionButton extends ImageButton {
+
+    public static final int SIZE_NORMAL = 0;
+    public static final int SIZE_MINI = 1;
```
Comment (davidvavra) [inline library/src/main/java/com/github/clans/fab/ExtendedFloatingActionButton.java]: We don't need two sizes for extended fab, remove.

File: library/src/main/java/com/github/clans/fab/ExtendedFloatingActionButton.java [hunk 3]
```diff
@@ -0,0 +1,1316 @@
...
+                label.onActionUp();
+            }
+            onActionUp();
+            return super.onSingleTapUp(e);
+        }
+    });
+    private boolean mUsingElevation;
+    private boolean mUsingElevationCompat;
+    // Progress
+    private boolean mProgressBarEnabled;
```
Comment (davidvavra) [inline library/src/main/java/com/github/clans/fab/ExtendedFloatingActionButton.java]: We don't need any progressbar in fab. Remove everything related to progress.

File: library/src/main/java/com/github/clans/fab/ExtendedFloatingActionButton.java [hunk 4]
```diff
@@ -0,0 +1,1316 @@
...
+            }
+        }
+
+        initShowAnimation(attr);
+        initHideAnimation(attr);
+        initReplaceExtendedAnimation();
+        initHideExtendedAnimation();
+        attr.recycle();
+
+        if (isInEditMode()) {
```
Comment (davidvavra) [inline library/src/main/java/com/github/clans/fab/ExtendedFloatingActionButton.java]: We don't need any EditMode. Remove everything related to editmode

File: library/src/main/java/com/github/clans/fab/FloatingActionMenu.java [hunk 5]
```diff
@@ -868,6 +734,14 @@ public void setIconAnimationCloseInterpolator(Interpolator closeInterpolator) {
         mCloseAnimatorSet.setInterpolator(closeInterpolator);
     }
 
+    public boolean isInCurrentPlace() {
```
Comment (davidvavra) [inline library/src/main/java/com/github/clans/fab/FloatingActionMenu.java]: Strange method name, when is it true? When fab is visible?

File: library/src/main/java/com/github/clans/fab/Util.java [hunk 6]
```diff
@@ -26,6 +26,10 @@ static boolean hasJellyBean() {
         return Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN;
     }
 
+    static boolean hasNougat() {
```
Comment (davidvavra) [inline library/src/main/java/com/github/clans/fab/Util.java]: You can remove this method now.

File: sample/src/main/res/layout/home_fragment.xml [hunk 7]
```diff
@@ -18,27 +18,60 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_gravity="bottom|right"
+        android:paddingBottom="20dp"
+        android:paddingRight="24dp"
+        android:src="@drawable/ic_drawing"
+        app:menu_backgroundColor="@color/black_a160"
+        app:menu_extendedButtonText="Add new expense"
+        app:menu_fab_label="MENU 2"
+        app:menu_isExtended="true">
```
Comment (davidvavra) [inline sample/src/main/res/layout/home_fragment.xml]: Remove this attribute, it will always be extended.

**Review by davidvavra — CHANGES_REQUESTED** [review]

File: library/src/main/java/com/github/clans/fab/FloatingActionMenu.java (hunk 5 above)
Comment (TomWayne98) [reply library/src/main/java/com/github/clans/fab/FloatingActionMenu.java]: I have already mentioned this method in app PR coment. I need it to check if the FAB should be in the curent place (f.e. tab) in the app. I usually use just visibility method to check it, but now it wasn't enough.

Comment (davidvavra) [reply library/src/main/java/com/github/clans/fab/FloatingActionMenu.java]: Please get rid of this method. You are passing activity state into the library and not doing anything with it in the library. If this is needed, create a state only in the activity.

**Pull request merged**